front/lib/resources/data_source_view_resource.ts

```diff
@@ -148,51 +207,59 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
   static async listForDataSourcesInVault(
     auth: Authenticator,
     dataSources: DataSourceResource[],
-    vault: VaultResource
+    vault: VaultResource,
+    fetchDataSourceViewOptions?: FetchDataSourceViewOptions
   ) {
-    return this.baseFetch(auth, {
+    return this.baseFetch(auth, fetchDataSourceViewOptions, {
       where: {
         dataSourceId: dataSources.map((ds) => ds.id),
         vaultId: vault.id,
       },
     });
   }
 
-  static async fetchById(auth: Authenticator, id: string) {
+  static async fetchById(
+    auth: Authenticator,
+    id: string,
+    fetchDataSourceViewOptions?: Omit<
+      FetchDataSourceViewOptions,
+      "limit" | "order"
+    >
+  ) {
     const fileModelId = getResourceIdFromSId(id);
     if (!fileModelId) {
       return null;
     }
 
-    const [dataSource] = await this.baseFetch(auth, {
-      where: {
-        id: fileModelId,
-      },
-    });
+    const [dataSource] = await this.baseFetch(
+      auth,
+      fetchDataSourceViewOptions,
+      {
+        where: {
+          id: fileModelId,
+        },
+      }
+    );
 
     return dataSource ?? null;
   }
 
   // Updating.
-  async updateParents(
+
+  private async update(
```

Comment:
We want to avoid generic method like this and would rather implement specific ones.

Reply:
this one is private and is here only to avoid repeating all the sequelize update stuff